### DISCUSSION TITLE
Generating a QR code with SkiaSharp

### DISCUSSION THREAD

**Discussion by jonsagara** [micjahn/ZXing.Net#493] [Q&A] [answered]
Hi,

I need to generate a QR code on the fly that is displayed on a web page using a base64 data string, i.e.:

`<img src="data:image/png;base64,aaaabbbbcc...." />`

Using `ZXing.Net`, `SkiaSharp`, and the `SkiaSharp` bindings, this is what I have come up with:

```csharp
const int widthPixels = 150;
const int heightPixels = 150;
const string qrCodeContent = "https://www.example.com";

BarcodeWriter qrWriter = new()
{
	Format = BarcodeFormat.QR_CODE,
	Options = new QrCodeEncodingOptions
	{
		Height = heightPixels,
		Width = widthPixels,
		Margin = 0,
	},
};

using SKBitmap qrCode = qrWriter.Write(qrCodeContent);
using SKImage skImage = SKImage.FromBitmap(qrCode);
using SKData encoded = skImage.Encode(SKEncodedImageFormat.Png, 100);
var imageData = Convert.ToBase64String(encoded.AsSpan());
```

This does generate a QR code that displays properly.

Are there any problems with this approach? Can it be improved or optimized in any way?

Thank you,

Jon

**Comment by micjahn** [comment, marked as answer]
Do you really need this line here?
`using SKImage skImage = SKImage.FromBitmap(qrCode);`
Isn't it possible to use qrCode directly?
`using SKData encoded = qrCode.Encode(SKEncodedImageFormat.Png, 100);`

**Reply by jonsagara** [reply]
Turns out I do not. Thanks for the feedback!
